cupy/linalg/norms.py

```diff
@@ -241,11 +241,15 @@ def slogdet(a):
     util._assert_nd_squareness(a)
 
     dtype = numpy.promote_types(a.dtype.char, 'f')
-    real_dtype = dtype
+    if dtype == numpy.complex64:
+        real_dtype = numpy.dtype("f")
+    elif dtype == numpy.complex128:
+        real_dtype = numpy.dtype("d")
+    else:
+        real_dtype = dtype
 
-    # TODO(kataoka): support complex types
-    if dtype not in (numpy.float32, numpy.float64):
-        msg = ('dtype must be float32 or float64'
+    if dtype not in ("f", "d", "F", "D"):
```

Comment:
Isn't `dtype` a dtype object instead of `str`?!
```suggestion
    if dtype not in (numpy.float32, numpy.float64,
                     numpy.complex64, numpy.complex128):
```